frontend/src/App.js

```diff
@@ -110,12 +101,34 @@ const BUILT_IN_MODES = [
   ["UNSURE", "Unsure"],
 ];
 
-function MainHeader(props) {
```

Comment:
I destructured the `props` here so that this is not needed inside the body of the function:

```
let datasetName = props.datasetName;
  let datasetInfo = props.datasetInfo;
  let categoryCounts = props.categoryCounts;
  let customModesByCategory = props.customModesByCategory;
  let categoryDispatch = props.categoryDispatch;

  let modelInfo = props.modelInfo;
  let setModelInfo = props.setModelInfo;

  let clusterIsOpen = props.clusterIsOpen;
  let setClusterIsOpen = props.setClusterIsOpen;
  let clusteringStrength = props.clusteringStrength;
  let selection = props.selection;
  let setSelection = props.setSelection;
  let clusters = props.clusters;
  let knnImagesDispatch = props.knnImagesDispatch;
  let setOrderingMode = props.setOrderingMode;
  let generateEmbedding = props.generateEmbedding;
  let setSubset = props.setSubset;
  let mode = props.mode;
  let setMode = props.setMode;
  let labelModeCategory = props.labelModeCategory;
```
I would do the same thing for other components inside this file, and then go further and put each component in its own file as this file is getting too long?